File: src/models/AdminModels.ts
// ───────────────── 공통 문자열 리터럴 타입 ─────────────────
export type Attendance = "출근전" | "출근" | "퇴근";
export type ConditionStatusKo = "좋음" | "불안" | "위험";
export type ConditionStatusEn = "GOOD" | "WARNING" | "DANGER";
export type ShippingStatus =
  | "배송대기"
  | "배송중"
  | "배송완료"
  | "CANCELED"
  | "PENDING"
  | "STARTED"
  | "DELIVERED";

// ───────────────── 인증 / 회원가입 ─────────────────
export interface AdminSignupData {
  id: number;
  email: string;
  role: string;
}

export interface AuthTokens {
  accessToken: string;
  refreshToken: string;
}

// ───────────────── 가입 대기자 ─────────────────
export interface PendingUser {
  id: number;
  name: string;
  phoneNumber: string;
  residence: string;
  licenseImage: string;
  birth: string;
  career: string;
  averageWorking: string;
  averageDelivery: string;
  bloodPressure: string;
  approvalStatus: boolean;
  role: string;
}

// ───────────────── 승인된 회원 ─────────────────
export interface ApprovedUser {
  userId: number;
  driverId: number;
  name: string;
  approvalStatus: boolean;
  profileImageUrl?: string | null;
  career: string;
  averageWorking: string;
  averageDelivery: string;
  height?: number;
  weight?: number;
  attendance: Attendance; // "출근전" | "출근" | "퇴근"
  residence: string;
  workTime?: string | null;
  deliveryStats?: string | null;
  conditionStatus: ConditionStatusKo;
}

// ───────────────── 기사 프로필 ─────────────────
export interface DriverProfile {
  driverId: number;
  name: string;
  phoneNumber: string;
  residence: string;
  regions: string[];
  conditionStatus: ConditionStatusKo | ConditionStatusEn;
  heartRate: number;
  stepCount: number;
  deliveryCount: number;
}

// ───────────────── 기사 배정 상품 ─────────────────
export interface DeliveryItem {
  productId: number;
  productName: string;
  recipientName: string;
  recipientPhoneNumber: string;
  address: string;
  detailAddress: string;
  postalCode: string;
  shippingStatus: ShippingStatus | string;
  driverId: number;
  deliveryImageUrl?: string | null;
}

// ───────────────── 심박수 타임라인 ─────────────────
export interface HeartRateTimelineItem {
  userId: number;
  driverName: string;
  heartRate: number;
  recordedAt: string; // ISO-8601
}

// ───────────────── 상품 타임라인 ─────────────────
export interface ProductTimelineItem {
  timelineId: number;
  productId: number;
  productName: string;
  status: "STARTED" | "DELIVERED" | "PENDING" | "CANCELED" | string;
  statusChangedAt: string;
  latitude: number | null;
  longitude: number | null;
  addressShort: string | null;
  driverName: string | null;
  driverPhoneNumber: string | null;
}

// ───────────────── 실시간 건강 데이터 ─────────────────
export interface RealtimeHealthItem {
  userId: string;
  step: number;
  heartRate: number;
  capturedAt: string;
}

// ───────────────── 실시간 위치(초기 로딩) ─────────────────
export interface LocationRealtimeItem {
  lat: number;
  lng: number;
  capturedAt: string; // ISO-8601
  addressShort: string;
  region: string; // 예: "구로구"
  timestamp: number; // epoch millis
}

// ───────────────── 기사 퇴근 후 건강 데이터  ─────────────────
export interface DriverHealthData {
  workTime: string;
  leaveWorkTime: string;
  finish1: string;
  finish2: string;
  finish3: string;
  step: number;
  heartRate: number;
  conditionStatus: ConditionStatusKo | string;
}

export interface DriverRealtimeData {
  driverId: number;
  step: number;
  heartRate: number;
  conditionStatus: string;
  modifiedDate: string;
}

// ───────────────── 관리자: 지역 배정  ─────────────────
export interface AssignRegionRequest {
  driverId: number;
  region1: string;
  region2: string;
}

export interface AssignRegionResponse {
  driverId: number;
  driverName: string;
  region1: string;
  region2: string;
}

File: src/models/admin_models.dart
// lib/models/admin_models.dart

/// 관리자 가입 성공 시 반환되는 data.id, email, role
class AdminSignupData {
  final int id;
  final String email;
  final String role;

  AdminSignupData({
    required this.id,
    required this.email,
    required this.role,
  });

  factory AdminSignupData.fromJson(Map<String, dynamic> json) {
    return AdminSignupData(
      id: json['id'] as int,
      email: json['email'] as String,
      role: json['role'] as String,
    );
  }
}

/// 로그인 후 토큰
class AuthTokens {
  final String accessToken;
  final String refreshToken;

  AuthTokens({
    required this.accessToken,
    required this.refreshToken,
  });

  factory AuthTokens.fromJson(Map<String, dynamic> json) {
    return AuthTokens(
      accessToken: json['accessToken'] as String,
      refreshToken: json['refreshToken'] as String,
    );
  }
}

/// 가입 대기자 항목
class PendingUser {
  final int id;
  final String name;
  final String phoneNumber;
  final String residence;
  final bool approvalStatus;
  final String birth;
  final String career;
  final String averageWorking;
  final String averageDelivery;
  final String bloodPressure;
  final String role;

  PendingUser({
    required this.id,
    required this.name,
    required this.phoneNumber,
    required this.residence,
    required this.approvalStatus,
    required this.birth,
    required this.career,
    required this.averageWorking,
    required this.averageDelivery,
    required this.bloodPressure,
    required this.role,
  });

  factory PendingUser.fromJson(Map<String, dynamic> json) {
    return PendingUser(
      id: json['id'] as int,
      name: json['name'] as String,
      phoneNumber: json['phoneNumber'] as String,
      residence: json['residence'] as String,
      approvalStatus: json['approvalStatus'] as bool,
      birth: json['birth'] as String,
      career: json['career'] as String,
      averageWorking: json['averageWorking'] as String,
      averageDelivery: json['averageDelivery'] as String,
      bloodPressure: json['bloodPressure'] as String,
      role: json['role'] as String,
    );
  }
}

/// 승인된 회원 항목
class ApprovedUser {
  final int id;
  final bool approvalStatus;
  final String profileImageUrl;
  final String name;
  final String attendance;
  final String residence;
  final String workTime;
  final String deliveryStats;
  final String conditionStatus;

  ApprovedUser({
    required this.id,
    required this.approvalStatus,
    required this.profileImageUrl,
    required this.name,
    required this.attendance,
    required this.residence,
    required this.workTime,
    required this.deliveryStats,
    required this.conditionStatus,
  });

  factory ApprovedUser.fromJson(Map<String, dynamic> json) {
    return ApprovedUser(
      id: json['id'] as int,
      approvalStatus: json['approvalStatus'] as bool,
      profileImageUrl: json['profileImageUrl'] as String,
      name: json['name'] as String,
      attendance: json['attendance'] as String,
      residence: json['residence'] as String,
      workTime: json['workTime'] as String,
      deliveryStats: json['deliveryStats'] as String,
      conditionStatus: json['conditionStatus'] as String,
    );
  }
}
